### ISSUE TITLE
As full rebuild widget, at change locale.

### ISSUE THREAD

**Issue by Dinver** [aleksanderwozniak/table_calendar#202]
Hi, i used flutter_translate. Sorry at stupid question, but how can full rebuild widget?

```
import 'package:flutter/material.dart';
import 'package:flutter_translate/flutter_translate.dart';
import 'package:reminder/localization/keys.dart';
import 'package:table_calendar/table_calendar.dart';

class TableCalendarHome extends StatefulWidget {

  State<TableCalendarHome> createState() => _TableCalendarHomeState();

}

class _TableCalendarHomeState extends State<TableCalendarHome> {

  CalendarController _calendarController;

  @override
  void initState() {
    super.initState();
    _calendarController = CalendarController();
  }

  Widget build(BuildContext context) {

    var _currentLocale = LocalizedApp.of(context).delegate.currentLocale.toString();

    return TableCalendar(
      calendarController: _calendarController,
      locale: _currentLocale,
      availableCalendarFormats: {
        CalendarFormat.month: translate(Keys.Datetime_Month),
        CalendarFormat.twoWeeks: translate(Keys.Datetime_Two_Week),
        CalendarFormat.week: translate(Keys.Datetime_Week)
      },
      startingDayOfWeek: _currentLocale == 'ru_RU' 
        ? StartingDayOfWeek.monday
        : StartingDayOfWeek.sunday,
    );
  }

  @override
  void dispose() {
    _calendarController.dispose();
    super.dispose();
  }

}
```
I get output:
![image](https://user-images.githubusercontent.com/36528841/75150750-1c111500-5716-11ea-86d7-275ed53d7130.png)

When changing the locale with out restart I get:

![image](https://user-images.githubusercontent.com/36528841/75150908-7e6a1580-5716-11ea-99f9-25fe8b3a8aba.png)

**availableCalendarFormats** language and **startingDayOfWeek** not change.

**Comment by AllanCoinaski** [comment]
I find it easier:

Initialize  app in main.dart:
  import 'package:intl/date_symbol_data_local.dart';
  initializeDateFormatting().then((_) => runApp(MyApp())

In TableCalendar:
locale: 'pt_BR', //your language
availableCalendarFormats: {
        CalendarFormat.month: 'mês', //translate...
        CalendarFormat.week: 'semana',
        CalendarFormat.twoWeeks: '2 semanas'
      },

**Comment by Dinver** [comment]
> I find it easier:
> 
> Initialize app in main.dart:
> import 'package:intl/date_symbol_data_local.dart';
> initializeDateFormatting().then((_) => runApp(MyApp())
> 
> In TableCalendar:
> locale: 'pt_BR', //your language
> availableCalendarFormats: {
> CalendarFormat.month: 'mês', //translate...
> CalendarFormat.week: 'semana',
> CalendarFormat.twoWeeks: '2 semanas'
> },

At initialize app, can set any locale. Problem at switch in run app. Calendar **locale** changing correctly. **availableCalendarFormats** and **startingDayOfWeek** remain the same, even using your method.

**Comment by aleksanderwozniak** [comment]
Hmm I will need to look into this. I'll let you know once progress has been made.

**Comment by shamilovtim** [comment]
@aleksanderwozniak I believe this might be related to what I was describing with the init method. I kept getting a calendar that didn't change over and over again even though the underlying state had changed.

**Comment by aleksanderwozniak** [comment]
Fixed on my local branch. This will be added in upcoming `3.0.0` release after I make some more improvements.

**Comment by aleksanderwozniak** [comment]
Resolved with version `3.0.0`.
